src/nlsolve/newton.jl

```diff
@@ -235,148 +324,143 @@ end
-        # not all predictors are uprev, for other forms of predictors, defined in u₀
-        if isdefined(integrator.cache, :u₀)
-            @.. broadcast=false ustep=integrator.cache.u₀ + z
+@inline function _compute_rhs!(tmp, ztmp, ustep, α, tstep, k,
+        invγdt, p, uprev, f::TF, z) where {TF<:DAEFunction}
+    @.. broadcast=false ztmp=(tmp + α * z) * invγdt
+    @.. ustep=uprev + z
+    f(k, ztmp, ustep, p, tstep)
+    return _vec(k), ustep
+end
+
+@inline function _compute_rhs!(tmp, ztmp, ustep, γ, α, tstep, k,
+        invγdt, method::MethodType, p, dt, f, z)
+    mass_matrix = f.mass_matrix
+    ustep = compute_ustep!(ustep, tmp, γ, z, method)
+    if method === COEFFICIENT_MULTISTEP
+        f(k, z, p, tstep)
+        if mass_matrix === I
+            @.. broadcast=false ztmp=tmp + k - (α * invγdt) * z
         else
-            @.. broadcast=false ustep=uprev + z
+            update_coefficients!(mass_matrix, ustep, p, tstep)
+            mul!(_vec(ztmp), mass_matrix, _vec(z))
+            @.. broadcast=false ztmp=tmp + k - (α * invγdt) * ztmp
         end
-        f(k, ztmp, ustep, p, tstep)
-        _vec(k)
     else
-        mass_matrix = integrator.f.mass_matrix
-        if nlsolver.method === COEFFICIENT_MULTISTEP
-            ustep = z
-            f(k, z, p, tstep)
-            if mass_matrix === I
-                @.. broadcast=false ztmp=tmp + k - (α * invγdt) * z
-            else
-                update_coefficients!(mass_matrix, ustep, p, tstep)
-                mul!(_vec(ztmp), mass_matrix, _vec(z))
-                @.. broadcast=false ztmp=tmp + k - (α * invγdt) * ztmp
-            end
+        f(k, ustep, p, tstep)
+        if mass_matrix === I
+            @.. ztmp=(dt * k - z) * invγdt
         else
-            @.. broadcast=false ustep=tmp + γ * z
-            f(k, ustep, p, tstep)
-            if mass_matrix === I
-                @.. broadcast=false ztmp=(dt * k - z) * invγdt
-            else
-                update_coefficients!(mass_matrix, ustep, p, tstep)
-                mul!(_vec(ztmp), mass_matrix, _vec(z))
-                @.. broadcast=false ztmp=(dt * k - ztmp) * invγdt
-            end
+            update_coefficients!(mass_matrix, ustep, p, tstep)
+            mul!(_vec(ztmp), mass_matrix, _vec(z))
+            @.. ztmp=(dt * k - ztmp) * invγdt
         end
-        _vec(ztmp)
     end
-    return b, ustep
+    return _vec(ztmp), ustep
 end
 
-@inline function _compute_rhs!(nlsolver::NLSolver{<:NLNewton, true, <:Array}, integrator,
-        f::TF, z) where {TF}
-    @unpack uprev, t, p, dt = integrator
-    @unpack tmp, ztmp, γ, α, cache = nlsolver
-    @unpack ustep, tstep, k, invγdt = cache
-    isdae = TF <: DAEFunction
-    b = if isdae
-        @inbounds @simd ivdep for i in eachindex(z)
-            ztmp[i] = (tmp[i] + α * z[i]) * invγdt
-        end
-        if isdefined(integrator.cache, :u₀)
+@inline function _compute_rhs!(tmp::Array, ztmp::Array, ustep::Array, α, tstep, k,
```

Comment:
Do we actually want to inline these?

Reply:
I'll remove them. They were just there because they used to be there. It is one where the function call cost is going to be way less than the linear solve, so I don't know why it was manually annotated in the first place.